### DISCUSSION TITLE
completions-at-point-functions

### DISCUSSION THREAD

**Discussion by david-jennings** [joaotavora/eglot#724] [Q&A] [answered]
I've got a fairly simple eglot installation that works for jumping to code references. I've just tried symbol completion and it seems not to be using the LSP. I looked at the setting of `completion-of-point-functions` and it's set to `(tags-completion-at-point-function)`. I was expecting to see it set to `eglot-completion-at-point`. Am I missing something?

**Comment by joaotavora** [comment, marked as answer]
> Am I missing something?

Yes, you probably are and your expectation is correct. You should post the contents of your `.emacs` or at least the parts related to Eglot. If it's very big to post here, you can bisect it looking for the culprit . 

In my Python buffers, for example, I see:

```
completion-at-point-functions is a variable defined in `minibuffer.el'.

Its value is
(eglot-completion-at-point python-completion-at-point t)
```

**Comment by david-jennings** [comment]
I think I have at least part of the answer. I just tried removing `flycheck` from `.emacs` (I'd been using that before I tried eglot) and that seems to have enabled both `eglot-rename` (which also wasn't working previously) and a more sensible `completion-at-point`.

Although if I look at `completion-at-point-functions` I still see:
```
completion-at-point-functions’s value is
(tags-completion-at-point-function)
```
This is the relevant part of my `.emacs`:
```
(when (locate-library "eglot")
  (add-hook 'python-mode-hook 'eglot-ensure)
  (add-hook 'c++-mode-hook 'eglot-ensure)
  (setq eglot-ignored-server-capabilites ; sic
	'(:documentHighlightProvider :hoverProvider))
  (add-hook 'eglot--managed-mode-hook (lambda() (flymake-mode -1)))
  )
```
Seems I previously tried to disable `flymake-mode` on a buffer-by-buffer basis.

**Reply by joaotavora** [reply]
```
(setq eglot-ignored-server-capabilites ; sic
```

Ugh!  Why did no-one tell me?? :facepalm:   Anyway thanks for spotting the typo 2 years down the line or sth.

**Reply by david-jennings** [reply]
Now that I look at `eglot.el` I see that it's using `flymake` itself, so my trying to disable it (for some reason) is probably wrong. I'll remove that and check again.

**Reply by joaotavora** [reply]
Yes, to your actual problem.  I would start with a empty .emacs, or at least with an .emacs void of Flymake, Flycheck, Lsp, company etc stuff.  Comment that out than re-add it slowly.

**Reply by david-jennings** [reply]
Yep. I think I messed it up by also loading `flycheck` and/or `flymake`.

Am I correct in assuming that `eglot` takes over the functionality of those two modules (using `flymake` behind the scenes)?

**Reply by joaotavora** [reply]
Yes, you are.  `M-x eglot` is a "manage it all for you"
